File: problems/3158-find-the-xor-of-numbers-which-appear-twice/solution.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public int duplicateNumbersXOR(int[] nums) {
        Map<Integer, Integer> counts = new HashMap<>();
        int xorSum = 0;

        for (int num : nums) {
            counts.put(num, counts.getOrDefault(num, 0) + 1);
        }

        for (Map.Entry<Integer, Integer> entry : counts.entrySet()) {
            if (entry.getValue() == 2) {
                xorSum ^= entry.getKey();
            }
        }

        return xorSum;
    }
}

File: problems/3158-find-the-xor-of-numbers-which-appear-twice/solution.py
class Solution:
    def duplicateNumbersXOR(self, nums: List[int]) -> int:
        counts = {}
        for num in nums:
            counts[num] = counts.get(num, 0) + 1
        
        xor_sum = 0
        for num, count in counts.items():
            if count == 2:
                xor_sum ^= num
        
        return xor_sum